Source/PluginProcessor.cpp

```diff
@@ -252,6 +251,34 @@ void PinkTromboneAudioProcessor::processBlock (AudioBuffer<float>& buffer, MidiB
 	}
 }
 
+void PinkTromboneAudioProcessor::timerCallback()

```

Comment:
Rather than do all this by creating another timer, I'd use an audio-rate ramp like the built in JUCE ADSR utility (https://docs.juce.com/master/classADSR.html) and using that to update the constrictions. You can downsample by drawing samples from the ramp on every tick, but only using them every 10 msec or whatever.